src/Symfony/Component/Cache/Traits/PdoTrait.php

```diff
@@ -192,13 +196,16 @@ public function prune()
      */
     protected function doFetch(array $ids)
     {
+        $connection = $this->getConnection();
+        $useDbalConstants = $connection instanceof Connection && class_exists(ParameterType::class);
+
         $now = time();
         $expired = [];
 
         $sql = str_pad('', (\count($ids) << 1) - 1, '?,');
         $sql = "SELECT $this->idCol, CASE WHEN $this->lifetimeCol IS NULL OR $this->lifetimeCol + $this->timeCol > ? THEN $this->dataCol ELSE NULL END FROM $this->table WHERE $this->idCol IN ($sql)";
-        $stmt = $this->getConnection()->prepare($sql);
-        $stmt->bindValue($i = 1, $now, \PDO::PARAM_INT);
+        $stmt = $connection->prepare($sql);
+        $stmt->bindValue($i = 1, $now, $useDbalConstants ? ParameterType::INTEGER : \PDO::PARAM_INT);
```

Comment:
Using the PDO constants on a DBAL 3 connection will break in PHP 8.1 because PDO changed the values of those constants. 😱